src/lib/icon/icon-registry.ts

```diff
@@ -33,12 +33,22 @@ export function getMdIconNoHttpProviderError(): Error {
                    'Please include the HttpModule from @angular/http in your app imports.');
 }
 
+
+/**
+ * Returns an exception to be thrown when a URL couldn't be sanitized.
+ * @param url URL that was attempted to be sanitized.
+ * @docs-private
+ */
+export function getMdIconFailedToSanitizeError(url: SafeResourceUrl): Error {
+  return new Error(`MdIconRegistry has failed to sanitize the following URL: ${url}.`);
```

Comment:
A more accurate error message would be something like "The url provided to MdIconRegistry was not trusted as a resource URL via Angular's DomSanitizer"